### ISSUE TITLE
Support for smart numbering when making numbered lists

### ISSUE THREAD

**Issue by tbdalgaard** [jgm/djot#45]
Is it possible to make support for the following syntax when writing lists:

```

1. This is the first list item.
#. This is the second item.
#. This is the third list item.
```

This could make numbering lists very nice, when moving list items around when editing a document.

**Comment by jgm** [comment]
You can already do
```
1. one
1. another
1. another
```
Only the first number in the sequence is significant.

**Comment by tbdalgaard** [comment]
Ok, I am just very used to the way Pandoc can show this, so I thought this syntax looks better, and I like that items can be sequenced like how Pandoc can do this. I understand that DJOT isn't Pandoc at all, I just thought this syntax is easier when reading and composing the document.

**Comment by jgm** [comment]
I'll consider it.

**Comment by leroycep** [comment]
Asciidoc also has this feature. It allows you to make ordered lists with just a period, like so:

```asciidoc
Ordered list:

. One
. Two
. Three
```

**Comment by mtsknn** [comment]
> Asciidoc [...] allows you to make ordered lists with just a period

One benefit of using just a period instead of `1.` or `#.`: proper indentation for the list item's body would be 2 spaces (nice) instead of 3 spaces (weird). But this syntax (just a period) would probably work only with the "1." list marker type.

**Comment by jgm** [comment]
If we do this I prefer `#.`, because `.` looks too much like a bullet list in the source.

**Comment by waldyrious** [comment]
Plus there's precedent for `#` as numbered list markup, e.g. in MediaWiki. Though I do wonder if it could be confusing to have such a similar syntax to that used for headings meaning something completely different when a period is involved.

**Comment by jgm** [comment]
I agree that there's some danger of visually confusing `#.` with a heading.

Of course there might be an argument for changing the heading syntax to use `=` instead of `#` -- that is also a strong tradition in light markup.

Would the people who want something like this be happy with `0.`?

**Comment by waldyrious** [comment]
> Would the people who want something like this be happy with `0.`?

I'd rather see `=` used for headings and reserving `#` for numbered lists. The practical downside is that the more we deviate from markdown, the harder the adoption of Djot will be, but on the other hand, besides MediaWiki and other lightweight markup languages using `=` for headings, even markdown supports the symbol associated with headings, particularly in setext-style headings.

**Comment by tbdalgaard** [comment]
I wouldn't mind making two spaces after the list marker like
#. {space {space}
I have never liked the use of = for headings. I think the \# in for defining headers is easier to read.
Note: I am one of the people who really loves the idea behind DJOT, but aren't smart enough to code anything... I just love to markup text with options like DJOT. :)

**Comment by hex4badass** [comment]
What about using `$` for an ordered list?

```
$ One
$ Two
$ Three
```
`$` is already used in Mathematical and numerical contexts (LaTeX math and currency).

**Comment by dereuromark** [comment]
> You can already do
> 
> ```
> 1. one
> 1. another
> 1. another
> ```
> 
> Only the first number in the sequence is significant.

With this already working, whats the incentive/motivation to use `$` char?
